src/main/java/mate.academy/lib/Injector.java

```diff
@@ -8,6 +19,60 @@
     }
 
     public Object getInstance(Class<?> interfaceClazz) {
-        return null;
+        Object clazzImplementationInstance = null;
+        Class<?> clazz = findImplementation(interfaceClazz);
+
+        if (!clazz.isAnnotationPresent(Component.class)) {
+            throw new RuntimeException(
+                    "Class " + clazz.getName() + " is not annotated with @Component"
+            );
+        }
+
+        Field[] declaredFields = clazz.getDeclaredFields();
+        for (Field field : declaredFields) {
+            if (field.isAnnotationPresent(Inject.class)) {
+                Object fieldInstance = getInstance(field.getType());
```

Comment:
The recursive call to `getInstance` can lead to a stack overflow if there are circular dependencies. Consider adding a mechanism to detect and handle circular dependencies.